chore(CI): Upload release Windows zips from GH actions workflow

anthonybilinski opened the pull request:
- [x] Commits follow our [git commit guidelines](https://github.com/qTox/qTox/blob/master/CONTRIBUTING.md#git-commit-guidelines)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/qtox/qtox/6434)
<!-- Reviewable:end -->

anthonybilinski committed:
chore(CI): Upload release Windows zips from GH actions workflow

sphaerophoria reviewed (commented):
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6434#-:-Mt-hNSH97YJ-dVUah7v:b-jbsf8z)* status: :shipit: complete! 1 of 1 LGTMs obtained

___
*[.github/workflows/test.yaml, line 249 at r1](https://reviewable.io/reviews/qtox/qtox/6434#-Mt-gwcM4X20KOC-wbO8:-Mt-gwcM4X20KOC-wbO9:b25x9vm) ([raw file](https://github.com/qtox/qtox/blob/6a4249fa156584bc1601fae41f2f38cf3349086c/.github/workflows/test.yaml#L249)):*
> ```YAML
>           name: setup-qtox-x86_64-${{ matrix.build_type }}.zip
>           path: install-prefix/qtox-x86_64-${{ matrix.build_type }}.zip
>           if-no-files-found: ignore
> ```

I guess we can probably remove if-no-files-found: ignore now right? Now that we're not hardcoding the debug zip this should always work


<!-- Sent from Reviewable.io -->

anthonybilinski committed:
chore(CI): Remove if-no-files-found from artifact upload

Artifacts are now conditionally uploaded based on matrix, so they should always be present when expected.

anthonybilinski reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/qtox/qtox/6434)* status: :shipit: complete! 1 of 1 LGTMs obtained

___
*[.github/workflows/test.yaml, line 249 at r1](https://reviewable.io/reviews/qtox/qtox/6434#-Mt-gwcM4X20KOC-wbO8:-Mt0jnrg4h8DkzkN1M_t:b-vlo6d9) ([raw file](https://github.com/qtox/qtox/blob/a07ab89cc84bcab3a75277dd4efd1608535127bb/.github/workflows/test.yaml#L249)):*
<details><summary><i>Previously, sphaerophoria wrote…</i></summary><blockquote>

I guess we can probably remove if-no-files-found: ignore now right? Now that we're not hardcoding the debug zip this should always work
</blockquote></details>

Yes you're right, done.


<!-- Sent from Reviewable.io -->
